agent/state_generator.py

```diff
@@ -23,6 +25,15 @@ def test(cls, board: Board, current_player: Color):
         for i in range(len(boards)):
             print(F"{i + 1} => {StateParser.convert_board_to_text(boards[i])}")
 
+    @staticmethod
+    def generate_random_move(board: Board, player: Color):
+        """
+        Searches valid moves for a board and chooses one at random.
+        :return: A random move.
+        """
+        moves = StateGenerator.enumerate_board(board, player)
+        return random.choice(moves)
+
```

Comment:
An approach I've seen around that might be worth investigating in more detail (not necessarily this week) is having a single abstract agent interface that gets subclassed by a bunch of different kinds of agent implementations. So theoretically you'd be able to sub in a random agent, depth-1 agent, depth-2 agent, IDDFS agent etc.